core/store/src/trie/state_parts.rs

```diff
@@ -176,34 +177,44 @@ impl Trie {
         Ok(())
     }
 
-    /// Returns the storage changes for the state part.
+    /// Returns the storage changes for the state part and all contract codes extracted from it.
     /// Writing all storage changes gives the complete trie.
     pub fn apply_state_part(
         state_root: &StateRoot,
         part_id: u64,
         num_parts: u64,
         part: Vec<Vec<u8>>,
-    ) -> Result<TrieChanges, StorageError> {
+    ) -> Result<(TrieChanges, Vec<ContractCode>), StorageError> {
         if state_root == &CryptoHash::default() {
-            return Ok(TrieChanges::empty(CryptoHash::default()));
+            return Ok((TrieChanges::empty(CryptoHash::default()), vec![]));
         }
-        let trie = Trie::from_recorded_storage(PartialStorage { nodes: PartialState(part) });
+        let trie =
+            Trie::from_recorded_storage(PartialStorage { nodes: PartialState(part.clone()) });
```

Comment:
why do you need clone here? I do not see any more usages of part

Reply:
Yeah, it is an artifact of debugging.